lib/organisms/gravity-form/gravity-form.service.js

```diff
@@ -1,146 +1,144 @@
-(function () {
-
-    /**
-     * Created by Chris on 12/04/2016.
-     */
-
-    'use strict';
-
-
-    function lnOGravityFormService($http, $cacheFactory, $q, $log, $lnOGravityFormConfig, lnOGravityFormGETFormTransformer) {
-        $log.debug('lnOGravityFormService ->', $lnOGravityFormConfig);
+angular
+    .module('lnPatterns')
+    .factory('lnOGravityFormService', lnOGravityFormService);
+
+/*@ngInject*/
+function lnOGravityFormService($http, $cacheFactory, $q, $log, $lnOGravityFormConfig, lnOGravityFormGETFormTransformer) {
+    $log.debug('lnOGravityFormService ->', $lnOGravityFormConfig);
+
+    function getFormParams(formId) {
+        var form = $lnOGravityFormConfig.forms[formId];
+        if (typeof form === "undefined") {
+            $log.error('lnOGravityFormService.formatAPIAuthParams() -> form ID: ', formId, ' not found in config!');
+            return {};
+        }
+        return form;
+    }
 
+    function formatAPIAuthParams(form) {
+        return {
+            api_key: $lnOGravityFormConfig.api_key,
+            signature: form.signature,
+            expires: form.expires
+        };
+    }
 
-        // Internal
-        //
+    function checkTrailingSlash(url) {
+        var _u = angular.copy(url);
+        if (_u.substr(-1) != '/') {
+            _u += '/';
+        }
+        return _u;
+    }
 
+    function query(method, endpoint, params, clearCache, configParams) {
 
-        /**
-         * NOTE: This - along with other Auth processes - would normally go in an Interceptor, but as
-         * this component is effectively sharing a module with a whole bunch of other components,
-         * we don't know what effect that might have in the future if other components need to make HTTP requests....
-         */
-        function formatAPIAuthParams(key, sig, expires) {
-            return '/?api_key=' + key + '&signature=' + sig + '&expires=' + expires;
+        if (!$lnOGravityFormConfig || !$lnOGravityFormConfig.isValid) {
+            $log.error("lnOGravityFormService.query() -> Please make sure Gravity Form API parameters have been set!");
+            return;
         }
 
-        function query(method, endpoint, params, clearCache, configParams, requiresAuth) {
-
-            if (!$lnOGravityFormConfig || !$lnOGravityFormConfig.isValid) {
-                alert("There is a problem! Please make sure Gravity Form API parameters have been set!");
-                return;
+        var _baseUrl = checkTrailingSlash($lnOGravityFormConfig.api_base);
+
+        if (clearCache === true) {
```

Comment:
The getForm method in the service sets the clearCache param to false for that reason :)